### PR TITLE
Improve transformer dev mode performance

### PR DESCRIPTION
**Pull request by stuartwdouglas**

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkus/runner/bootstrap/StartupActionImpl.java [hunk 1]
```diff
@@ -66,12 +72,47 @@ public StartupActionImpl(CuratedApplication curatedApplication, BuildResult buil
...
+        handleEagerClasses(runtimeClassLoader, eagerClasses);
+    }
+
+    private void handleEagerClasses(QuarkusClassLoader runtimeClassLoader, Set<String> eagerClasses) {
+        int availableProcessors = Runtime.getRuntime().availableProcessors();
+        if (availableProcessors == 1) {
+            return;
+        }
+        //leave one processor for the main startup thread
+        ExecutorService loadingExecutor = Executors.newFixedThreadPool(availableProcessors - 1);
```
Comment (gsmet) [inline core/deployment/src/main/java/io/quarkus/runner/bootstrap/StartupActionImpl.java:84]: What if you have only one core in a container?

**Review by gsmet — CHANGES_REQUESTED** [review]
Added a somewhat stupid question.

I saw you marked it for backport, is it safe enough?

File: core/deployment/src/main/java/io/quarkus/runner/bootstrap/StartupActionImpl.java (hunk 1 above)
Comment (stuartwdouglas) [reply core/deployment/src/main/java/io/quarkus/runner/bootstrap/StartupActionImpl.java:84]: This code is not run, there is a check 4 lines above that will bail out.

Comment (gsmet) [reply core/deployment/src/main/java/io/quarkus/runner/bootstrap/StartupActionImpl.java:84]: Hum. Silly me.

**Review by gsmet — APPROVED** [review]

**Pull request merged**